app/pokedex/pokedex-client.tsx

```diff
@@ -0,0 +1,35 @@
+"use client";
+import { useState } from "react";
+import PokedexTile from "./pokedex-tile";
+import Search from "../../components/search";
+import { Pokemon } from "../../types/pokemon";
+import "../../styles/pokedex/pokedex-tile.css";
+
+interface PokedexClientProps {
+  pokemonList: Pokemon[];
+  error: string | null;
+}
+
+export default function PokedexClient({ pokemonList, error }: PokedexClientProps) {
+  const [search, setSearch] = useState("");
+  const [list] = useState<Pokemon[]>(pokemonList);
+
+  const filteredList = list.filter((pokemon) =>
+    pokemon.name.toLowerCase().includes(search.toLowerCase())
+  );
+
+  return (
```

Comment:
improvement could show a message if no pokemon match the search criteria

Reply:
Good call, updated